dpxdt/client/workers.py

```diff
@@ -67,13 +71,22 @@ def _print_tree(obj):
     def _get_dict_for_repr(self):
         return self.__dict__
 
-    def __repr__(self):
+    def _print_repr(self, depth):
+        if depth <= 0:
+            return '%s.%s#%d' % (
+                self.__class__.__module__,
+                self.__class__.__name__,
+                id(self))
+        
         return '%s.%s(%s)#%d' % (
             self.__class__.__module__,
             self.__class__.__name__,
-            self._print_tree(self._get_dict_for_repr()),
+            self._print_tree(self._get_dict_for_repr(), depth - 1),
```

Comment:
Can you add a comment here with the rationale for the depth? Would be nice to have your note about stack depth.

Over 80 chars on this line.